### DISCUSSION TITLE
Server Error RangeError: Invalid time value

### DISCUSSION THREAD

**Discussion by devnuru** [vercel/next.js#25426] [Help]
Hey Guys, How can I solve this issue?

Server Error
RangeError: Invalid time value

This error happened while generating the page. Any console logs will be displayed in the terminal window.
Source
.next\server\pages\posts\[id].js (5:45) @ Date

  3 | export default function Date({ dateString }) {
  4 |   const date = parseISO(dateString)
> 5 |   return <time dateTime={dateString}>{format(date, 'LLLL d, yyyy')}</time>
    |                                             ^
  6 | }
Call Stack
Function.getInitialProps
.next\server\pages\_document.js (70:33)

**Comment by Texian** [comment]
I had this problem, too, and in short:
[id].js is trying to turn EVERY file into a post, including itself, which naturally returns an invalid value. So, move all the blog post .md files into their own folder and redirect the code to look in that new folder and the problem should resolve.

In longer:
I’m not sure where I missed this step either, but somewhere along the lines there was a bit of a reorganization of the file structure and it throws everything for a loop. Plus, the constant use of “posts” without variation, including two discrete folders being named “posts” leads to some confusion and semantic satiation. 
The general file structure by the end of that lesson as far as the relevant files are concerned (folders in bold) is:

**topFolder**
— **lib**
—— posts.js
— **pages**
—— **posts**
——— [id].js
—— index.js
— **posts**
—— pre-rendering.md
—— ssg-ssr.md

So, in posts.js make sure the path.join() function is pointing to the correct folder with the markdown files.
Move the markdown files into a top-level folder of their own.
In [id].js make sure both functions are async and “const postData = **await** getPostData(params.id)
In index.js that getStaticProps is an async function.
And if all else fails, the api-routes-starter repo is all set up properly and can be used for reference: https://github.com/vercel/next-learn/tree/master/basics/api-routes-starter

It’s over a year later, but hopefully this’ll help you and others who might run into the same problem.

**Comment by NafiurSiddiqui** [comment]
For some reason, no matter what I do, including following @Texian's instructions, The error is always there. Looks like every year someone this patient makes it to this page.  

If anyone else just wanna get unstuck, you can use the following Vanilla JS code for the time being. 

```
export default function FormattedDate({ dateString }) {
	const [month, day, year] = dateString.split('-');
	const date = new Date(year, month - 1, day);
	const monthName = date.toLocaleString('default', { month: 'long' });

	return (
		<time dateTime={dateString}>
			{monthName} {day}, {year}
		</time>
	);
}
```

simply import it inside `[id]` and use it as `<FormattedDate  dateString={ postData.date} />` . Works like a charm!
